examples/bench/echoserver.py

```diff
@@ -74,6 +74,21 @@ def data_received(self, data):
         self.transport.write(data)
 
 
+class EchoBufferedProtocol(asyncio.BufferedProtocol):
+    def connection_made(self, transport):
+        self.transport = transport
+        self.buffer = bytearray(256 * 1024)
```

Comment:
```python
self._buffer = bytearray(256 * 1024)
self.buffer = memoryview(self._buffer)
```

Reply:
Ha! Been there, should have commented - to make sure the echo is true, the buffer needs to be copied, so that next read doesn’t change previous outgoing data which is still pending. With memoryview and bytes again, the benchmark gets slightly slower, therefore it ended up like this now. Thanks anyway! I’ll add comment.

Reply:
Ah, right :)

We'll need to add new API to asyncio 3.8 to tell us when a `write()` has completed so that we can re-use the buffer.

Reply:
That’s interesting!